proxycore/cluster.go

```diff
@@ -33,33 +33,65 @@ const (
 	DefaultRefreshTimeout = 5 * time.Second
 )
 
+type Event interface {
```

Comment:
Added a small, unrelated refactor here to not use `interface{}` for events.

Reply:
Nice